src/equations/linearized_euler_2d.jl

```diff
@@ -0,0 +1,174 @@
+"""
+struct LinearizedEulerEquations2D{RealT<:Real} <: AbstractLinearizedEulerEquations{2, 4}
+    rho_mean::RealT
+    v1_mean::RealT
+    v2_mean::RealT
+    c::RealT
+  
+    function LinearizedEulerEquations2D(rho_mean, v1_mean, v2_mean, c)
+        if rho_mean < 0
+          throw(ArgumentError("rho_mean must be non-negative"))
+        elseif c < 0
+          throw(ArgumentError("c must be non-negative"))
+        end
+        
+        return new{typeof(rho_mean)}(rho_mean, v1_mean, v2_mean, c)
+    end
+end
+    
+function LinearizedEulerEquations2D(; rho_mean, v1_mean, v2_mean, c)
+    return LinearizedEulerEquations2D(rho_mean, v1_mean, v2_mean, c)
+end
+    
+    
+varnames(::typeof(cons2cons), ::LinearizedEulerEquations2D) = ("rho_prime", "v1_prime", "v2_prime", "p_prime")
+varnames(::typeof(cons2prim), ::LinearizedEulerEquations2D) = ("rho_prime", "v1_prime", "v2_prime", "p_prime")
+    
+"""
+    initial_condition_convergence_test(x, t, equations::LinearizedEulerEquations2D)
+    
+A smooth initial condition used for convergence tests.
+"""
+function initial_condition_convergence_test(x, t, equations::LinearizedEulerEquations2D)
+    rho_prime = -cospi(2*t) * (sinpi(2*x[1]) + sinpi(2*x[2]))
+    v1_prime = sinpi(2*t) * cospi(2*x[1])
+    v2_prime = sinpi(2*t) * cospi(2*x[2])
+    p_prime = rho_prime
+      
+    return SVector(rho_prime, v1_prime, v2_prime, p_prime)
+end
+    
+    
+"""
+    boundary_condition_wall(u_inner, orientation, direction, x, t, surface_flux_function,
+                                equations::LinearizedEulerEquations2D)
+    
+Boundary conditions for a solid wall.
+"""
+function boundary_condition_wall(u_inner, orientation, direction, x, t, surface_flux_function,
+                                    equations::LinearizedEulerEquations2D)
+    # Boundary state is equal to the inner state except for the velocity. For boundaries
+    # in the -x/+x direction, we multiply the velocity in the x direction by -1.
+    # Similarly, for boundaries in the -y/+y direction, we multiply the velocity in the
+    # y direction by -1
+    if direction in (1, 2) # x direction
+        u_boundary = SVector(u_inner[1], -u_inner[2], u_inner[3], u_inner[4])
+    else # y direction
+        u_boundary = SVector(u_inner[1], u_inner[2], -u_inner[3], u_inner[4])
+    end
+      
+    # Calculate boundary flux
+    if iseven(direction) # u_inner is "left" of boundary, u_boundary is "right" of boundary
+        flux = surface_flux_function(u_inner, u_boundary, orientation, equations)
+    else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+        flux = surface_flux_function(u_boundary, u_inner, orientation, equations)
+    end
+      
+    return flux
+end
+    
+    
+# Calculate 1D flux for a single point
+@inline function flux(u, orientation::Integer, equations::LinearizedEulerEquations2D)
+    (; rho_mean, v1_mean, v2_mean, c) = equations
```

Comment:
```suggestion
    @unpack rho_mean, v1_mean, v2_mean, c = equations
```
This is imho a little easier to read and follows what we use elsewhere in Trixi (but this is just a suggestion). There are also a few other locations in this file to which this applies.